compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -64,6 +75,21 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
 
                 rhs_then_block.unit()
             }
+            ExprKind::LogicalOp { op: LogicalOp::Or, lhs, rhs } => {
+                let lhs = &this.thir[lhs];
+                let rhs = &this.thir[rhs];
+                let (lhs_then_block, lhs_else_block) =
+                    this.test_atom(lhs, this.source_info(lhs.span), temp_scope_override, block);
```

Comment:
Going further: is there a way to use `then_else_break` here too, and put the `rhs` as the target for the break?

Reply:
That's a good point. Currently I am pursuing this rewrite.